File: composePeakasso/findRectangles.py
# Python program to find all  
# rectangles filled with 0 
  
def findend(i,j,a,output,index): 
    x = len(a) 
    y = len(a[0]) 
  
    # flag to check column edge case, 
    # initializing with 0 
    flagc = 0
  
    # flag to check row edge case, 
    # initializing with 0 
    flagr = 0
  
    for m in range(i,x):  
  
        # loop breaks where first 1 encounters 
        if a[m][j] == 1:  
            flagr = 1 # set the flag 
            break
  
        # pass because already processed 
        if a[m][j] == 5:  
            pass
  
        for n in range(j, y):  
  
            # loop breaks where first 1 encounters 
            if a[m][n] == 1: 
                flagc = 1 # set the flag 
                break
  
            # fill rectangle elements with any 
            # number so that we can exclude 
            # next time 
            a[m][n] = 5
  
    if flagr == 1: 
        output[index].append( m-1) 
    else: 
        # when end point touch the boundary 
        output[index].append(m)  
  
    if flagc == 1: 
        output[index].append(n-1) 
    else: 
        # when end point touch the boundary 
        output[index].append(n)  
  
  
def get_rectangle_coordinates(a): 
  
    # retrieving the column size of array 
    size_of_array = len(a)  
  
    # output array where we are going 
    # to store our output  
    output = []  
  
    # It will be used for storing start 
    # and end location in the same index 
    index = -1
  
    for i in range(0,size_of_array): 
        for j in range(0, len(a[0])): 
            if a[i][j] == 0: 
  
                # storing initial position  
                # of rectangle 
                output.append([i, j])  
  
                # will be used for the  
                # last position 
                index = index + 1        
                findend(i, j, a, output, index)  
  
  
    print (output) 
  
# driver code 
tests = [ 
  
            [1, 1, 1, 1, 1, 1, 1], 
            [1, 1, 1, 1, 1, 1, 1], 
            [1, 1, 1, 0, 0, 0, 1], 
            [1, 0, 1, 0, 0, 0, 1], 
            [1, 0, 1, 1, 1, 1, 1], 
            [1, 0, 1, 0, 0, 0, 0], 
            [1, 1, 1, 0, 0, 0, 1], 
            [1, 1, 1, 1, 1, 1, 1] 
  
        ] 
  
  
get_rectangle_coordinates(tests)

File: composePeakasso/findRectangles.c
#include <stdio.h>
#include <stdlib.h>

void findend(int i, int j, char** canvas, int output[100][4], int counter) {
    int x = sizeof(canvas)/sizeof(canvas[0]);
    int y = sizeof(canvas[i])/sizeof(canvas[i][0]);

    int flagc = 0;
    int flagr = 0;
    int m, n;
    for (m = i; m < x; m++) {
        if (canvas[m][j] == ' ') {
            flagr = 1;
            break;
        }
        if (canvas[m][j] == 5) {
            continue;
        }

        for (n = j; n < y; n++) {
            if (canvas[m][n] == '*') {
                flagc = 1;
                break;
            }
            canvas[m][n] = 5;
        }
    }
    if (flagr == 1) {
        output[counter][2] = (m-1);
    } else {
        output[counter][2] = (m);
    }

    if (flagc == 1) {
        output[counter][3] = (n-1);
    } else {
        output[counter][3] = (n);
    }
}


void get_rectangle_coordinates(int x, int y, char** canvas) {

    int output[100][4];
    int counter = 0;

    for( int i = 0; i < sizeof(canvas)/sizeof(canvas[0]); i++) {
        for (int j = 0; j < sizeof(canvas[i])/sizeof(canvas[i][0]); j++) {
            if (canvas[i][j] == '*') {
                output[counter][0] = i;
                output[counter][1] = j;
                findend(i, j, canvas, output, counter);
                counter++;
            }
        }
    }
    for (int i = 0; i < counter; i++)
        printf("%d %d %d %d \n", output[i][0], output[i][1], output[i][2], output[i][3]);
}

int main() {
    int x, y;
    scanf("%d %d\n", &x, &y);
    char** canvas = (char**)malloc(sizeof(char) * y);
    for (int i = 0; i < y; i++) {
        canvas[i] = (char*)malloc(sizeof(char) * x);
    }
    char dump1, dump2;

    for( int i = 0; i < sizeof(canvas)/sizeof(canvas[0]); i++) {
        for (int j = 0; j < sizeof(canvas[i])/sizeof(canvas[i][0]); j++) {
            scanf("%c", &canvas[i][j]);
        }
        scanf("%c%c", &dump1, &dump2);
    }
    get_rectangle_coordinates(x, y, canvas);
}